### PR TITLE
Bug 1107696 - Fix not-in-range problem in Wifi panel, r=crh0716

### PR DESCRIPTION
**Pull request by EragonJ**
...ection, known network appears as "not in range"

### CODE REVIEW

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8d7b1ff2f3f9)

File: apps/settings/js/modules/wifi_utils.js [hunk 1]
```diff
@@ -250,6 +253,57 @@ define(function(require) {
...
+      }
+
+      if (listItem) {
+        listItem.classList.add('active');
+        listItem.querySelector('small').
+          setAttribute('data-l10n-id', 'shortStatus-' + networkStatus);
+        if (networkStatus === 'connecting') {
+          listItem.querySelector('aside').classList.add('connecting');
+        }
+        if (networkStatus === 'connected') {
```
Comment (crh0716) [inline apps/settings/js/modules/wifi_utils.js]: nit: `else if`

File: apps/settings/js/modules/wifi_utils.js [hunk 2]
```diff
@@ -250,6 +253,57 @@ define(function(require) {
...
+          listItem.querySelector('aside').classList.remove('connecting');
```
Comment (crh0716) [inline apps/settings/js/modules/wifi_utils.js]: Should we also add 'connected'?

File: apps/settings/js/modules/wifi_utils.js [hunk 3]
```diff
@@ -250,6 +253,57 @@ define(function(require) {
...
+      if (!network) {
+        return '';
+      } else {
+        var key =
+          network.ssid + '+' + WifiHelper.getSecurity(network).join('+');
+        return key;
+      }
+    },
+
+    updateListItemStatus: function(options) {
```
Comment (crh0716) [inline apps/settings/js/modules/wifi_utils.js]: Please add jsdoc for this method.

File: apps/settings/js/modules/wifi_utils.js [hunk 4]
```diff
@@ -250,6 +253,57 @@ define(function(require) {
...
+      var listItems = options.listItems;
+      var activeItem = options.activeItem;
+      var event = options.event;
```
Comment (crh0716) [inline apps/settings/js/modules/wifi_utils.js]: To make the function clearer, I would suggest to use `options.network` and `options.status`. For others who want to reuse this function they won't need to figure out what the event object really is.

File: apps/settings/js/panels/wifi_join_hidden/panel.js [hunk 5]
```diff
@@ -36,8 +36,8 @@ define(function(require) {
       onHide: function() {
         isHomeKeyPressed = document.hidden;
         if (!isHomeKeyPressed) {
-          elements.password = '';
-          elements.identity = '';
+          elements.password.value = '';
```
Comment (crh0716) [inline apps/settings/js/panels/wifi_join_hidden/panel.js]: !!!

File: apps/settings/js/modules/wifi_utils.js [hunk 6]
```diff
@@ -250,6 +253,57 @@ define(function(require) {
...
+
+      if (!event || !listItems) {
```
Comment (crh0716) [inline apps/settings/js/modules/wifi_utils.js]: If we always need `event` and `listItems`, I would suggest not to include them in `options`.

File: apps/settings/js/modules/wifi_utils.js (hunk 1 above)
Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: ok !

File: apps/settings/js/panels/wifi_join_hidden/panel.js (hunk 5 above)
Comment (EragonJ) [inline apps/settings/js/panels/wifi_join_hidden/panel.js]: !!!

File: apps/settings/js/modules/wifi_utils.js (hunk 3 above)
Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: ok !

File: apps/settings/js/modules/wifi_utils.js (hunk 4 above)
Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: good idea.

File: apps/settings/js/modules/wifi_utils.js (hunk 6 above)
Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: I prefer to use `options` way to make the interface more consistent.

Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: Here, we can do early return to gracefully degrade if needed elements are not passing in and would use `console.log` to print some log informations.

File: apps/settings/js/modules/wifi_utils.js (hunk 2 above)
Comment (EragonJ) [inline apps/settings/js/modules/wifi_utils.js]: okay !

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=86ba7dbdee75)

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=626cfaa2c79c)

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3d88d76c309b)

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f6040d8f949b)

**Pull request merged**